lib/src/arithmetic/floating_point/floating_point_adder_simple.dart

```diff
@@ -17,27 +17,31 @@ class FloatingPointAdderSimple<FpType extends FloatingPoint>
   /// Add two floating point numbers [a] and [b], returning result in [sum].
   /// - [adderGen] is an adder generator to be used in the primary adder
   /// functions.
-  /// - [priorityGen] is a [PriorityEncoder] generator to be used in the
-  /// leading one detection (default [RecursiveModulePriorityEncoder]).
   FloatingPointAdderSimple(super.a, super.b,
       {super.clk,
       super.reset,
       super.enable,
       Adder Function(Logic a, Logic b, {Logic? carryIn}) adderGen =
           NativeAdder.new,
-      PriorityEncoder Function(Logic bitVector, {Logic? valid, String name})
-          priorityGen = RecursiveModulePriorityEncoder.new,
+      List<int> Function(int) widthGen =
+          CarrySelectCompoundAdder.splitSelectAdderAlgorithmSingleBlock,
       super.name = 'floatingpoint_adder_simple'})
       : super(
             definitionName: 'FloatingPointAdderSimple_'
                 'E${a.exponent.width}M${a.mantissa.width}') {
+    if (a.runtimeType != b.runtimeType) {
```

Comment:
this seems like a dangerous check because it doesn't consider any sort of class polymorphism.  e.g. what if someone extends an FP32 type to have some additional functionality that wouldn't break any functionality in this adder?

I think better would be to check things that definitely matter (e.g. exponent and mantissa widths) and then document the behavior clearly (e.g. "always uses input a's version of so-and-so functionality", if something like that exists).

Reply:
match on explicitJBit boolean.